src/superphot_plus/samplers/licu_sampler.py

```diff
@@ -0,0 +1,219 @@
+from superphot_plus.posterior_samples import PosteriorSamples
+from superphot_plus.samplers.sampler import Sampler
+from superphot_plus.surveys.fitting_priors import MultibandPriors
+from superphot_plus.utils import calculate_neg_chi_squareds
+
+__all__ = ["LiCuSampler"]
+
+
+class LiCuSampler(Sampler):
+    """Fit light curves using the light-curve package's VillarFit
+
+    Parameters
+    ----------
+    **licu_kwargs : dict
+        Keyword arguments to pass to the light-curve package's VillarFit,
+        most notably 'algorithm', which can be 'ceres', 'lmsder', 'mcmc',
+        'mcmc-ceres' and 'mcmc-lmsder'. Default is 'ceres'. Please also
+        consider setting 'mcmc_niter' which is the number of MCMC iterations
+        defaults to 128.
+        Boundaries and initial guesses are set by this code, so you don't
+        need to set them yourself.
+    """
+
+    def __init__(self, **licu_kwargs):
+        kwargs = {'algorithm': 'ceres'}
+        kwargs.update(licu_kwargs)
+        self.licu_kwargs = kwargs
+
+    def run_single_curve(
+        self, lightcurve: Lightcurve, priors: MultibandPriors, **kwargs
+    ) -> PosteriorSamples:
+        """Perform model fitting using light-curve on a single light curve.
+
+        This function runs the dynesty importance nested sampling algorithm
+        on a single light curve.
+
+        Parameters
+        ----------
+        lightcurve : Lightcurve object
+            The light curve of interest.
+        priors : MultibandPriors object
+            The priors to use for fitting.
+        Returns
+        -------
+        samples: PosteriorSamples
+            Return the MCMC samples or None if the fitting is
+            skipped or encounters an error.
+        """
+        max_flux, max_flux_loc = lightcurve.find_max_flux()
+        reference_idx = priors.ref_band_index
+        priors_clip_a, priors_clip_b, priors_mean, _priors_std = (
+            transform_priors_to_physical_values(a, max_flux, max_flux_loc, reference_idx=reference_idx)
+            for a in priors.to_numpy().T
+        )
+
+        cube = []
+        for band, clip_a, clip_b, mean in zip(
+                priors.ordered_bands,
+                priors_clip_a.reshape(-1, 7),
+                priors_clip_b.reshape(-1, 7),
+                priors_mean.reshape(-1, 7)
+        ):
+            lc = lightcurve.filter_by_band(band, in_place=False)
+            cube.extend(fit_single_band(lc, clip_a, clip_b, mean, **self.licu_kwargs))
+        cube = np.asarray(cube)
+
+        # Scale parameters to the reference band
+        non_ref_idx = np.delete(np.arange(len(priors.ordered_bands)), reference_idx)
+        param_matrix = cube.reshape(-1, 7)
+        param_matrix[non_ref_idx] /= param_matrix[reference_idx]
+        # Fix for out of the limits parameters and for NaNs
+        cube = np.clip(cube, priors_clip_a, priors_clip_b)
+        cube = np.where(np.isnan(cube), priors_mean, cube)
```

Comment:
this works, but if you want to preserve fit chisq maybe default out-of-bounds values to the closest bound instead of the mean